src/NexusMods.CrossPlatform/Process/XDGOpenDependency.cs

```diff
@@ -0,0 +1,68 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using CliWrap;
+
+namespace NexusMods.CrossPlatform.Process;
+
+internal class XDGOpenDependency : ExecutableRuntimeDependency
+{
+    public override string DisplayName => "xdg-open";
+    public override string Description => "Open a URL in the user's preferred application that handles the respective URL or file type.";
+    public override Uri Homepage { get; } = new("https://www.freedesktop.org/wiki/Software/xdg-utils/");
+    public override OSPlatform[] SupportedPlatforms { get; } = [OSPlatform.Linux];
+
+    public XDGOpenDependency(IProcessFactory processFactory) : base(processFactory) { }
+
+    public Command CreateOpenUriCommand(Uri uri)
+    {
+        return Cli.Wrap("xdg-open").WithArguments([uri.ToString()], escape: true);
+    }
+
+    protected override Command BuildQueryCommand(PipeTarget outpuPipeTarget)
+    {
+        var command = Cli.Wrap("xdg-open").WithArguments("--version").WithStandardOutputPipe(outpuPipeTarget);
```

Comment:
Technically it's not needed, but would it be helpful to also consume stderr and stdin in the ExecutableRuntimeDependenc(ies) like in the ProcessFactory just in case?

My main concern is someone copying the code from here, and launching something that doesn't properly handle an invalid stderr/stdin pipe, like the issue we had with Steam on Linux.

Reply:
This just builds the command, it doesn't execute it. All command execution should go through the process factory, which handles IO correctly.

Reply:
Ah, you're right, I forgot that I made it handle the generic case. That's my bad.